Bottom: 6944f9d613c7aeb15bd34e968c05b34d3dfb26d7
Top:    94c330d035f3a8b015955420b24857ae9e4b4457
Author: Ian Kent <raven@themaw.net>
Date:   2013-06-07 12:38:47 +0800

autofs-5.0.7 - add pgrp check in do_spawn()

The process group should never be negative here but add a check anyway.


---

diff --git a/daemon/spawn.c b/daemon/spawn.c
index 9b8d5a2..abb353a 100644
--- a/daemon/spawn.c
+++ b/daemon/spawn.c
@@ -201,7 +201,8 @@ static int do_spawn(unsigned logopt, unsigned int wait,
 
 			seteuid(0);
 			setegid(0);
-			setpgid(0, pgrp);
+			if (pgrp >= 0)
+				setpgid(0, pgrp);
 		}
 
 		execv(prog, (char *const *) argv);
